cypress/e2e/tables-unified-search.cy.js

```diff
@@ -1,14 +1,21 @@
 let localUser
 let localUser2
+const groupId = (Math.random() + 1).toString(36).substring(7)
 
 describe('The Home Page', () => {
 
 	before(function() {
-		cy.createRandomUser().then(user => {
-			localUser = user
-		})
-		cy.createRandomUser().then(user => {
-			localUser2 = user
+		cy.createGroup(groupId).then(() => {
+			cy.createRandomUser().then(user => {
+				localUser = user
+			}).then(user => {
+				cy.addUserToGroup(user.userId, groupId)
+			})
+			cy.createRandomUser().then(user => {
+				localUser2 = user
+			}).then(user => {
+				cy.addUserToGroup(user.userId, groupId)
+			})
 		})
```

Comment:
I don't think you need the `.then()` after `cy.createGroup(groupId)`. It's implicit as cypress will wait for the command to finish if it returns a promise. If it does not then adding `.then(...)` also won't help.

So here's a little less nested version:
```suggestion
		cy.createGroup(groupId)
		cy.createRandomUser().then(user => {
			localUser = user
		}).then(user => {
			cy.addUserToGroup(user.userId, groupId)
		})
		cy.createRandomUser().then(user => {
			localUser2 = user
		}).then(user => {
			cy.addUserToGroup(user.userId, groupId)
		})
```

Reply:
Bonus exercise ;)... If you make use of `localUser` and `localUser2` you can also get rid of the `then` function around the calls to `cy.addUserToGroup`.